src/libs/actions/Link.js

```diff
@@ -49,8 +51,13 @@ function openOldDotLink(url) {
 
         const params = _.compact([authTokenParam, emailParam]).join('&');
 
-        // If the URL contains # or ?, we can assume they don't need to have the `?` token to start listing url parameters.
-        return `${CONFIG.EXPENSIFY.EXPENSIFY_URL}${url}${hasHashParams || hasURLParams ? '&' : '?'}${params}`;
+        return Environment.getOldDotEnvironmentURL()
+            .then((environmentURL) => {
+                const oldDotDomain = Url.addTrailingForwardSlash(environmentURL || CONFIG.EXPENSIFY.EXPENSIFY_URL);
```

Comment:
1. Is it possible for `environmentURL` to be `null | undefined` here? Why?
2. Is it OK to fallback to the production address? If yes, why not make `getOldDotEnvironmentURL()` have the fallback

Reply:
> Is it possible for environmentURL to be null | undefined here? Why?

Usually not. We can remove it. I added just for a safety. I see promise is always resolved.

Reply:
If for some reason `getOldDotEnvironmentURL` doesn't work as expected, the fallback here would hide the bug instead of point it out

Reply:
Agree, I can't think of a case where it would fail. I am going to remove it.